api/client/src/main/java/javax/websocket/ContainerProvider.java

```diff
@@ -39,17 +41,18 @@
      * @return an implementation provided instance of type WebSocketContainer
      */
     public static WebSocketContainer getWebSocketContainer() {
-         WebSocketContainer wsc = null;
-        for (ContainerProvider impl : ServiceLoader.load(ContainerProvider.class)) {
-            wsc = impl.getContainer();
-            if (wsc != null) {
-                return wsc;
-            } 
-        }
-        if (wsc == null) {
-            throw new RuntimeException("Could not find an implementation class.");
-        } else {
+        Iterator<ContainerProvider> providers = ServiceLoader.load(ContainerProvider.class).iterator();
+        if (providers.hasNext()) {
```

Comment:
So we can differentiate between ServiceLoader.load() returning an empty iteration (no implementations available) and all returned implementations returning null for getContainer(). The original code was trying to differentiate between these two but not doing so. Some of the code was unreachable.
There are ways this could be written with a for loop but this was cleaner.

Reply:
I understand the purpose of your code now. Thanks.

Reply:
What do we want to do if the iterator gives us say ...

1. Bad Entry that fails
2. Good Entry that works

with the implementation as you have provided in this PR the `getWebSocketContainer()` will fail.
The [javadoc for `ServiceLoader.iterator()`](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html#iterator--) gives some advice on at least catching  `ServiceConfigurationError` and skipping bad entries.

Reply:
The behaviour in this respect is unchanged with the PR. If getContainer() throws an uncaught exception then it gets passed to the caller. I was trying to limit this PR to fixing the code so a) there was no longer any unreachable code and b) the code behaves originally intended.

I'm not against making the code more robust along the lines you suggest but it raises a number of questions that, I think, are worthy of a separate discussion (exactly what to catch and what to re-throw, do we try and log any exceptions that are swallowed, is it better to fail and let the caller see the exception, etc.)

Reply:
The we defer this point to discussion on issue #254